src/core/qgspointlocator.cpp

```diff
@@ -661,6 +665,12 @@ void QgsPointLocator::setExtent( const QgsRectangle *extent )
   destroyIndex();
 }
 
+void QgsPointLocator::setRenderContext( const QgsRenderContext &context )
+{
+  mContext = std::unique_ptr<QgsRenderContext>( new QgsRenderContext( context ) );
+
+  destroyIndex();
+}
```

Comment:
It seems to me there is no way to turn off visibility-dependent behavior of point locator after it has been enabled?

Reply:
It is possible to configure the behavior in the preferences and then rebuild the index

Reply:
> Why is it necessary to destroy the index? 

It's necessary to destroy the index, if you change the style of your layer, since the subject is especially to filter invisible feature.
Initially there was a bug when you checked a "sub style". I added possible cases.
In use, only styleChanged is really needed. What do you think if I do this:

remove:
`
connect( mLayer, &QgsVectorLayer::rendererChanged, this, &QgsPointLocator::destroyIndex );
connect( mLayer, &QgsVectorLayer::styleChanged, this, &QgsPointLocator::destroyIndex );
connect( mLayer, &QgsVectorLayer::layerModified, this, &QgsPointLocator::destroyIndex );
`

add `connect( mLayer, &QgsVectorLayer::styleChanged, this, &QgsPointLocator::destroyIndex );` in `QgsPointLocator::setRenderContext`